serde_v8/src/ser.rs

```diff
@@ -430,11 +432,24 @@ impl<'a, 'b, 'c> ser::SerializeMap for MapSerializer<'a, 'b, 'c> {
 
 pub struct Serializer<'a, 'b, 'c> {
   scope: ScopePtr<'a, 'b, 'c>,
+  intern_strings: bool,
 }
 
 impl<'a, 'b, 'c> Serializer<'a, 'b, 'c> {
   pub fn new(scope: ScopePtr<'a, 'b, 'c>) -> Self {
-    Serializer { scope }
+    Serializer {
+      scope,
+      intern_strings: false,
+    }
+  }
+  pub fn with_string_interning(
```

Comment:
`interning` is the act of working as an intern, this should perhaps be `with_string_internalizing`

Reply:
Ops. That makes total sense.

That said, I think string internalization on map keys might be too workload specific. So maybe it's best left out of this PR, or implemented on the rust side.

Reply:
> Ops. That makes total sense.
> 
> That said, I think string internalization on map keys might be too workload specific. So maybe it's best left out of this PR, or implemented on the rust side.

Yes, I was going to point that out.

Internalizing struct field strings is fine since the set is finite and relatively small (for deno's ops), internalizing all Map keys gives users a footgun which can lead to unbounded memory "leaks". I would prefer this to be removed from this PR